src/components/ga4/EventBuilder/ValidateEvent/useValidateEvent.ts

```diff
@@ -125,44 +133,54 @@ const useValidateEvent = (): Requestable<
     }
     setStatus(RequestStatus.InProgress)
     setValidationMessages([])
-    let validatorErrors = useFirebase ? validatePayloadAttributes(payload) : []
-
-    validateHit(payload, instanceId, api_secret)
-      .then(messages => {
-        setTimeout(() => {
-          if (messages.length > 0 || validatorErrors.length > 0) {
-            let apiValidationErrors = messages.filter(a =>
-                a.fieldPath === "measurement_id"
-                  ? !useFirebase
-                  : a.fieldPath === "firebase_app_id"
-                  ? useFirebase
-                  : true
-              )
-            
-            apiValidationErrors.forEach(err => {
-              if (!validatorErrors.map(e => e.description).includes(err.description)) {
-                validatorErrors.push(err)
-              }
-            })
-
-            validatorErrors = formatErrorMessages(validatorErrors, payload)
-            console.log('validatorErrors', validatorErrors)
-            
-            setValidationMessages(validatorErrors)
-            setStatus(RequestStatus.Failed)
-          } else {
-            setStatus(RequestStatus.Successful)
-          }
-        }, 250)
-      })
+
+    if (!useTextBox || Object.keys(payload).length !== 0) {
+      let validatorErrors = useFirebase ? validatePayloadAttributes(payload) : []
+
+      validateHit(payload, instanceId, api_secret)
+        .then(messages => {
+          setTimeout(() => {
+            if (messages.length > 0 || validatorErrors.length > 0) {
+              let apiValidationErrors = messages.filter(a =>
+                  a.fieldPath === "measurement_id"
+                    ? !useFirebase
+                    : a.fieldPath === "firebase_app_id"
+                    ? useFirebase
+                    : true
+                )
+              
+              apiValidationErrors.forEach(err => {
+                if (!validatorErrors.map(e => e.description).includes(err.description)) {
+                  validatorErrors.push(err)
+                }
+              })
+
+              validatorErrors = formatErrorMessages(validatorErrors, payload)
+              
+              setValidationMessages(validatorErrors)
+              setStatus(RequestStatus.Failed)
+            } else {
+              setStatus(RequestStatus.Successful)
+            }
+          }, 250)
+        })
       .catch(e => {
         console.error(e)
       })
-  }, [status, payload, api_secret, instanceId, useFirebase])
+    } else {
+      let validatorErrors = formatValidationMessage()
+      setValidationMessages(validatorErrors)
+      setStatus(RequestStatus.Failed)
+    }
+  }, [status, payload, api_secret, instanceId, useFirebase, payloadErrors])
```

Comment:
Could you please take a look?

"React Hook useCallback has missing dependencies: 'useTextBox' and 'validatePayloadAttributes'. Either include them or remove the dependency array"

Reply:
Done!